src/Compilers/Core/Portable/Compilation/OptimizationLevel.cs

```diff
@@ -48,42 +48,46 @@ public enum OptimizationLevel
 
     internal static class OptimizationLevelFacts
     {
+        internal static (OptimizationLevel OptimizationLevel, bool DebugPlus) DefaultValues { get; } = (OptimizationLevel.Debug, false);
+
         public static string ToPdbSerializedString(this OptimizationLevel optimization, bool debugPlusMode)
-        => optimization switch
-        {
-            OptimizationLevel.Release => "release",
-            OptimizationLevel.Debug => debugPlusMode ? "debug-plus" : "debug",
-            _ => throw ExceptionUtilities.UnexpectedValue(optimization)
-        };
+            => (optimization, debugPlusMode) switch
+            {
+                (OptimizationLevel.Release, true) => "release-debug-plus",
```

Comment:
It feels like the original code assumed that debugPlusMode would always be false. It's unfortunate we didn't assert that assumption. Glad you found the problem here.